internal/webhooks/handler.go

```diff
@@ -0,0 +1,191 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/signalfx/splunk-otel-collector-operator/apis/o11y/v1alpha1"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=ignore,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io,sideEffects=none,admissionReviewVersions={v1,v1beta1}
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;watch
+// +kubebuilder:rbac:groups=o11y.splunk.com,resources=splunkotelagents,verbs=get;list;watch
+// +kubebuilder:rbac:groups="apps",resources=replicasets,verbs=get;list;watch
+
+const (
+	operatorNamespace           = "splunk-otel-operator-system"
+	envSplunkOtelAgent          = "SPLUNK_OTEL_AGENT"
+	envOTELServiceName          = "OTEL_SERVICE_NAME"
+	envOTELExporterOTLPEndpoint = "OTEL_EXPORTER_OTLP_ENDPOINT"
+	envOTELTracesExporter       = "OTEL_TRACES_EXPORTER"
+	envOTELResourceAttrs        = "OTEL_RESOURCE_ATTRIBUTES"
+
+	exporterOTLP   = "otlp"
+	exporterJaeger = "jaeger-thrift-splunk"
+
+	annotationConfInjectionEnabled = "o11y.splunk.com/inject-config"
+)
+
+type handler struct {
+	client  client.Client
+	logger  logr.Logger
+	decoder *admission.Decoder
+}
+
+type config struct {
+	exporter string
+	endpoint string
+}
+
+// NewHandler creates a new WebhookHandler.
+func NewHandler(logger logr.Logger, cl client.Client) admission.Handler {
+	return &handler{
+		client: cl,
+		logger: logger,
+	}
+}
+
+func (h *handler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	pod := corev1.Pod{}
+	err := h.decoder.Decode(req, &pod)
+	if err != nil {
+		h.logger.Error(err, "unable to decode pod")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if !strings.EqualFold(pod.Annotations[annotationConfInjectionEnabled], "true") {
```

Comment:
So it's applied only on pods with annotation `o11y.splunk.com/inject-config=true` right? Reading the PR comment it sounds like it's being applied by default

Reply:
That was the original implementation but I changed it to be opt in. Will update description.

Reply:
I think going forward we would need support of namespace attributes as well